pkg/controllers/nodeclaim/lifecycle/registration.go

```diff
@@ -88,13 +92,35 @@ func (r *Registration) syncNode(ctx context.Context, nodeClaim *v1beta1.NodeClai
 	// Sync all taints inside NodeClaim into the Node taints
 	node.Spec.Taints = scheduling.Taints(node.Spec.Taints).Merge(nodeClaim.Spec.Taints)
 	node.Spec.Taints = scheduling.Taints(node.Spec.Taints).Merge(nodeClaim.Spec.StartupTaints)
+	// Remove karpenter.sh/unregistered taint
+	taints, err := isolateKarpenterStartupTaint(node.Spec.Taints)
+	if err != nil {
+		return fmt.Errorf("missing required startup taint: %w", err)
```

Comment:
Should we do the check on this taint at the top of the sub-reconcile (right after we do the condition check)? This would allow us to short-circuit so all we have to do is remove the taint in the `syncNode` call.

Reply:
Checking after getting node from nodeclaim and before syncing the node.